flamegraph.pl

```diff
@@ -682,16 +685,21 @@ sub flow {
 		}
 		
 		t.textContent = txt;
+
 		// Fit in full text width
-		if (/^ *\$/.test(txt) || t.getSubStringLength(0, txt.length) < w)
+		if (/^ *\$/.test(txt) || t.getComputedTextLength() < w)
```

Comment:
I'm finding that `t.getComputedTextLength`, like `g.getSubStringLength`, takes a long time when `txt` is long, 5200 characters in my test.  It takes a few seconds in Chrome 49.0.2612.0.

I've been working on solving a similar problem in `s(node)` when hovering over an element with a long `title`.  I'm still working on it, but I've found that starting with a short string, then expanding it until I get to the max width performs much better, e.g.

```
function s(node) { // show
   info = g_to_text(node);
   details.nodeValue = info.substr(0, 100);
   var maxWidth = details.parentElement.parentElement.getBBox().width - details.parentElement.getBBox().x;
   var infoLength = info.length;
   while (details.nodeValue.length < infoLength && details.parentElement.getSubStringLength(0, infoLength) < maxWidth) {
      details.nodeValue = info.substr(0, details.nodeValue.length + 1);
   }
}
```